ocp_resources/data_source.py

```diff
@@ -39,19 +40,29 @@ def to_dict(self):
                 },
             })
 
-    @property
-    def pvc(self):
-        data_source_pvc = self.instance.spec.source.pvc
-        pvc_name = data_source_pvc.name
-        pvc_namespace = data_source_pvc.namespace
+    def _get_boot_source(self, boot_source_type):
```

Comment:
You can get the resource class here directly.
and use it like: `self._get_boot_source(boot_source_type=PersistentVolumeClaim)`
No need for line 50 when calling it like this.

Reply:
Hi ^^
Ye though about it,
but we need to get the boot source from the Datasource instance, so we will need to use "pvc" or "snapshot" like in line 44 in order to get its name & namespace

So we can all the function with the resource class like you suggested, but we will still need a condition for line 44 ,
and I found it easier to pass string to the function

wdyt?

Reply:
Missed that line, in this case what you doing here is OK.